### PR TITLE
Addon-docs: Refactor docs support into individual framework packages

### PR DESCRIPTION
**Pull request by shilman**
Issue: #16210

## What I did

- [x] Created a new library `@storybook/docs-tools` to contain common functions
- [x] Moved framework into `angular`, `ember`, `html`, `react`, `svelte`, `vue`, `vue`, `web-components`
- [x] Cleaned up `addon-docs` complex preset system 

In the telescoping PR https://github.com/storybookjs/storybook/pull/17697
- [x] Added the ability to disable docs support for these packages
- [x] Updated the documentation

## What to review

1. Naming: does `docs-tools` make sense?
2. Overall structure: does the framework update with the `preset-framework-docs`, which in turn calls into `docs-tools` make sense? Does this bloat the framework packages in some way that I'm missing, other than what's handled in #17697?

## How to test

I tested this using Storybook test-runner, doing visual snapshot tests against the docs pages in both `next` and this branch.

You can see the feature/instructions here: https://github.com/storybookjs/test-runner/pull/79

### CODE REVIEW

File: addons/docs/src/frameworks/common/config.ts [hunk 1]
```diff
@@ -1,12 +1,8 @@
-import { enhanceArgTypes } from './enhanceArgTypes';
-
 export const parameters = {
   docs: {
     inlineStories: false,
     getContainer: async () => (await import('../../blocks')).DocsContainer,
     getPage: async () => (await import('../../blocks')).DocsPage,
-    iframeHeight: 100,
+    iframeHeight: 110,
```
Comment (yannbf) [inline addons/docs/src/frameworks/common/config.ts:6]: Any particular reason for this change?

Comment (shilman) [reply addons/docs/src/frameworks/common/config.ts:6]: Nope, good catch. I was trying to debug something and wanted a more unique number. Will change it back!

Comment (shilman) [inline addons/docs/src/frameworks/common/config.ts:6]:
```suggestion
    iframeHeight: 100,
```

File: lib/docs-tools/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Storybook Docs Utils
+
+Shared utility functions for frameworks to implement docs
+
+- ArgType extraction
+- Dynamic snippet generation
```
Comment (tmeasday) [inline lib/docs-tools/README.md:6]: Perhaps add a comment to this file outlining that it depended on by all the frameworks and should be as minimal as possible.

**Review by tmeasday — COMMENTED** [review]
LGTM

File: lib/docs-tools/README.md (hunk 2 above)
Comment (shilman) [reply lib/docs-tools/README.md:6]: Good point. Added in the other PR

**Pull request merged**